media/js/newsletter/management.es6.js

```diff
@@ -797,7 +804,11 @@ const NewsletterManagementForm = {
                         NewsletterManagementForm.onDataError(e);
                     });
             })
-            .catch(NewsletterManagementForm.redirectToRecoveryPage);
+            .catch(() => {
+                if (!FormUtils.getUserToken()) {
+                    NewsletterManagementForm.redirectToRecoveryPage();
```

Comment:
Is this an extra safety belt added while in the neighbourhood?

Reply:
Right now if there's any kind of JS error thrown during the data fetching process, we'll catch the error silently and then redirect to the recover page. Just just makes it so we only redirect back we know we don't have a valid token, which makes debugging easier.